test/powershell/Modules/Microsoft.PowerShell.Utility/Sort-Object.Tests.ps1

```diff
@@ -21,204 +21,946 @@
+        }
+    }
+
+    It 'Return the bottom N integers sorted in ascending order' {
+        $unsortedData = 973474993,271612178,-1258909473,659770354,1829227828,-1709391247,-10835210,-1477737798,1125017828,813732193
+        $baseSortParameters = @{}
+        $fullSortResults = $unsortedData | Sort-Object @baseSortParameters
+        $bottomNSortResults = $unsortedData | Sort-Object @baseSortParameters -Bottom 3
+        $offset = $fullSortResults.Count - $bottomNSortResults.Count
+        for ($i = 0; $i -lt $bottomNSortResults.Count; $i++) {
+            $bottomNSortResults[$i] | Should Be $fullSortResults[$i + $offset]
+        }
+    }
+
+    It 'Return the bottom N integers sorted in descending order' {
+        $unsortedData = 973474993,271612178,-1258909473,659770354,1829227828,-1709391247,-10835210,-1477737798,1125017828,813732193
+        $baseSortParameters = @{Descending=$true}
+        $fullSortResults = $unsortedData | Sort-Object @baseSortParameters
+        $bottomNSortResults = $unsortedData | Sort-Object @baseSortParameters -Bottom 3
+        $offset = $fullSortResults.Count - $bottomNSortResults.Count
+        for ($i = 0; $i -lt $bottomNSortResults.Count; $i++) {
+            $bottomNSortResults[$i] | Should Be $fullSortResults[$i + $offset]
+        }
+    }
+
+    It 'Return the top N objects from a collection of different types of objects, sorted in ascending order' {
+        $unsortedData = @(
+            ($item0 = 'x')
+            ($item1 = Get-Alias where)
+            ($item2 = 'c')
+            ($item3 = ([pscustomobject]@{}) | Add-Member -Name ToString -Force -MemberType ScriptMethod -Value {$null} -PassThru) # Use this syntax to get $null from a default sort (uses ToString)
+            ($item4 = 42)
+            ($item5 = Get-Alias foreach)
+            ,($item6 =@('a','b','c')) # Use this syntax to pass an array with a few values to sort-object (also useful when testing sorts with unexpected data)
+            ($item7 = [pscustomobject]@{Name='NotAnAlias'})
+            ($item8 = [pscustomobject]@{Name=$null;Definition='Custom'})
+            ($item9 = 'z')
+            ,($item10 = ,$null) # Use this syntax to pass an array with a single $null value to sort-object (also useful when testing sorts with unexpected data)
+        )
+        $baseSortParameters = @{}
+        $fullSortResults = $unsortedData | Sort-Object @baseSortParameters
+        $topNSortResults = $unsortedData | Sort-Object @baseSortParameters -Top 6
+        for ($i = 0; $i -lt $topNSortResults.Count; $i++) {
+            if ($topNSortResults[$i] -is [System.Array]) {
+                $topNSortResults[$i].Equals($fullSortResults[$i]) | Should Be $true
+            } else {
+                $topNSortResults[$i] | Should Be $fullSortResults[$i]
+            }
+        }
+    }
+
+    It 'Return the top N objects from a collection of different types of objects, sorted in descending order' {
+        $unsortedData = @(
```

Comment:
Actually, after a closer look, these tests are a good example of why `Pester` has a parameter `-TestCases` to `It`. Please consider changing the tests to use `-TestCases`, I think it will simplify these test quite a bit.